packages/modal/src/ux-modal-service.ts

```diff
@@ -0,0 +1,402 @@
+      }
+      return true; // this allow normal behvior with this click for other purposes
+    };
+  }
+
+  public addLayer(modal: UxModal, bindingContext: UxModalBindingContext) {
+    const layerCount = this.modalLayers.push({
+      bindingContext,
+      modal
+    });
+    if (layerCount === 1) {
+      this.setListener();
+    }
+  }
+
+  public getLayer(modal: UxModal): UxModalLayer | undefined {
+    const index = this.modalLayers.map(i => i.modal).indexOf(modal);
+    if (index !== -1) {
+      return this.modalLayers[index];
+    }
+    return undefined;
+  }
+
+  public removeLayer(modal: UxModal) {
+    const index = this.modalLayers.map(i => i.modal).indexOf(modal);
+    if (index !== -1) {
+      this.modalLayers.splice(index, 1);
+    }
+    if (this.modalLayers.length === 0) {
+      this.removeListener();
+    }
+  }
+
+  private setListener() {
+    window.addEventListener('keyup', this);
+    document.addEventListener('click', this.handleDocumentClick);
+  }
+
+  private removeListener() {
+    window.removeEventListener('keyup', this);
+    document.removeEventListener('click', this.handleDocumentClick);
+  }
+
+  public handleEvent(event: KeyboardEvent) {
+    const key = this.getActionKey(event);
+    if (key === undefined) {
+      return;
+    }
+    const activeLayer = this.getLastModal();
+    if (activeLayer === null) {
+      return;
+    }
+    const keyboard = activeLayer.keyboard;
+    if (key === 'Escape'
+      && (keyboard === true || keyboard === key || (Array.isArray(keyboard) && keyboard.indexOf(key) > -1))) {
+      activeLayer.dismiss();
+    } else if (key === 'Enter' && (keyboard === key || (Array.isArray(keyboard) && keyboard.indexOf(key) > -1))) {
+      activeLayer.ok();
+    }
+  }
+
+  private getActionKey(event: KeyboardEvent): UxModalKeybord | undefined {
+    if ((event.code || event.key) === 'Escape' || event.keyCode === 27) {
+      return 'Escape';
+    }
+    if ((event.code || event.key) === 'Enter' || event.keyCode === 13) {
+      return 'Enter';
+    }
+    return undefined;
+  }
+
+  public getLastLayer(): UxModalLayer | null {
+    return this.modalLayers.length > 0 ? this.modalLayers[this.modalLayers.length - 1] : null;
+  }
+
+  public getLastModal(): UxModal | null {
+    const lastLayer = this.getLastLayer();
+    return lastLayer !== null ? lastLayer.modal : null;
+  }
+
+  public get zIndex() {
+    return this.startingZIndex + this.modalLayers.length;
+  }
+
+  private createModalElement(options: UxModalServiceOptions, bindingContext: UxModalBindingContext): HTMLElement {
+    const element = document.createElement('ux-modal');
+    element.setAttribute('dismiss.trigger', 'dismiss()');
+    element.setAttribute('ok.trigger', 'ok($event)');
+    if (options.position !== undefined) {
+      element.setAttribute('position', options.position);
+    }
+    if (options.moveToBodyTag === false) {
+      element.setAttribute('move-to-body-tag.bind', 'false');
```

Comment:
- do we need the `-tag` suffix?
- I couldn't find this `move-to-body-tag` in `ux-model.ts`, is this needed? What is the default behavior for this?

Reply:
This whole `move-to-body-tag` is not required anymore. 

I had renamed the method `moveToHost()` and forgot to change here. But then I also removed the option to decide to move it or not. In fact it's handled by the `host` property => if set, we move the modal to its host without choice.

Thanks for catching this !